azavea/modellab-geoprocessing
# Add More Color Breaks

# First Commit

Breaks for NVDI, VARI, and VGI have been added.

Connects #35
## Testing Instructions
- Make sure that the code builds
- Run the jar, preferably on your host so that you can see the output from it on the console; Confirm that NVDI, VGI, and VARI are loaded.
- Compare the NVDI, VGI, and VARI files with the breaks given in azavea/modellab#151 to make sure that they were copied correctly.
# Second Commit

A "coerce" operation has been added to allow floating-point rasters to result from integral raster inputs.

Connects #37
## Testing Instructions

Choose one of the following:
### Choice 1

Post json blob to the server:

```
{
    "name": "LocalDivide",
    "breaks_name": "ndvi",
    "inputs": [{
        "name": "LocalSubtract",
        "inputs": [{
            "name": "LoadLayer",
            "layer_name": "us-pennsylvania-philadelphia-landsat-nir-30m-epsg3857",
            "layer_alias": "Landsat NIR",
            "hash": "COZ373Y"
        }, {
            "name": "LoadLayer",
            "layer_name": "us-pennsylvania-philadelphia-landsat-red-30m-epsg3857",
            "layer_alias": "Landsat Red",
            "hash": "3BWIPNI"
        }]
    }, {
        "name": "LocalAdd",
        "inputs": [{
            "name": "Coerce",
            "inputs": [{
                "name": "LoadLayer",
                "layer_name": "us-pennsylvania-philadelphia-landsat-nir-30m-epsg3857",
                "layer_alias": "Landsat NIR",
                "hash": "COZ373Y"
            }]
        }, {
            "name": "LoadLayer",
            "layer_name": "us-pennsylvania-philadelphia-landsat-red-30m-epsg3857",
            "layer_alias": "Landsat Red",
            "hash": "3BWIPNI"
        }]
    }]
}
```

Visit [this endpoint](http://localhost:8888/sample/MSSM45Q/12/-75.13893127441406/39.97554154865437?bufferSize=2) and verify that there are floating-point results.  (The assumption is that you started the code using `spark-submit` on your host.)
### Choice 2

Test along with the solution for azavea/modellab#191.
# Third Commit

The operation name in echoed structure input would be e.g. "LocalLocalUnarySqrt" prior to this change, now it is just "LocalUnarySqrt".

Connects #34
